cmd/dockerd/tracing.go

```diff
@@ -82,13 +84,15 @@ func setupTracing(ctx context.Context) sdktrace.SpanProcessor {
 	case "http/protobuf", "":
 		exp, err = otlptracehttp.New(ctx)
 	default:
-		log.G(ctx).WithField("proto", proto).Error("Unsupported otlp protocol, only grpc and http/protobuf are supported, skipping")
-		return nil
+		return nil, errors.New("unsupported otlp protocol, only grpc and http/protobuf are supported")
 	}
 
 	if err != nil {
-		log.G(ctx).WithError(err).Error("Failed to create otlp exporter, skipping")
-		return nil
+		return nil, errors.Wrap(err, "failed to create otlp exporter")
+	}
+
+	if _, ok := os.LookupEnv(otelServiceNameEnv); !ok {
+		os.Setenv(otelServiceNameEnv, filepath.Base(os.Args[0]))
```

Comment:
Is this the "identifier" for the service? If so, would it make sense to use the daemon's ID for this? (instead of the file-path)?

Reply:
Yes, that could work. This is what I do for the integration test daemons, actually (by setting the env var).

Reply:
Updated to include the daemon ID.